Add the ability to define a seperate slug for a repository

sd-lueckel opened the pull request:
Ability to define a slug instead of using the name. If no slug is defined the name will be used (current implementations should not be affected)

Fixes #4

Commit:
Add the ability to define a seperate slug for a repository instead of using the name

grubernaut commented:
Hey @sd-lueckel, thanks for the contribution! 
Mind adding some acceptance tests for this change, as well as commenting with the test output? Thanks!

Commit:
Only set slug if it is not empty and if is different from repo.Name

Commit:
Add test for a camelcase repo with slug

Commit:
Added documentation

sd-lueckel commented:
Hi

I added an acceptance test and here are the test outputs. Also i updated the documentation

make test:
`==> Checking that code complies with gofmt requirements...
go test -i $(go list ./... |grep -v 'vendor') || exit 1
echo $(go list ./... |grep -v 'vendor') | \
                xargs -t -n4 go test  -timeout=30s -parallel=4
go test -timeout=30s -parallel=4 github.com/terraform-providers/terraform-provider-bitbucket github.com/terraform-providers/terraform-provider-bitbucket/bitbucket
?       github.com/terraform-providers/terraform-provider-bitbucket     [no test files]
ok      github.com/terraform-providers/terraform-provider-bitbucket/bitbucket   0.017s`

make testacc:
`==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-bitbucket     [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccBitbucketDefaultReviewers_basic
--- PASS: TestAccBitbucketDefaultReviewers_basic (7.50s)
=== RUN   TestAccBitbucketHook_basic
--- PASS: TestAccBitbucketHook_basic (11.78s)
=== RUN   TestAccBitbucketRepository_basic
--- PASS: TestAccBitbucketRepository_basic (4.36s)
=== RUN   TestAccBitbucketRepository_camelcase
--- PASS: TestAccBitbucketRepository_camelcase (3.88s)
PASS
ok      github.com/terraform-providers/terraform-provider-bitbucket/bitbucket   27.537s`

grubernaut reviewed (approved):
Awesome work here! Thanks!
